### ISSUE TITLE
Allow application/json in addition to custom mime type?

### ISSUE THREAD

**Issue by PandaWhisperer** [pandastrike/pbx#11]
I know there's probably also a good answer for this. Currently, PBX only accepts incoming JSON (i.e. for PUT and POST requests) with the exact custom media type it automatically generates. Any other media type, including `application/json` is answered with a `415 Unsupported Media Type`. 

Is it a good idea to be less restrictive about this? Should there be an option to do so?

**Comment by PandaWhisperer** [comment]
I noticed that it is possible to explicitly override the custom type by specifying the `type` parameter on a given action (`get`, `post`, or `put`), e.g.

``` coffeescript
builder.define "test"
.post type: "application/json"
```

This will create a path `/test` that will accept `POST` requests with a `Content-Type` of `application/json`. 

I suppose that renders this a non-issue, but since it wasn't totally obvious, perhaps we could add documentation for this.

**Comment by PandaWhisperer** [comment]
So unfortunately, due to #18, using the `type` parameter isn't quite optimal either. In fact, it's a rather terrible idea, because it breaks the schema validation in a nasty and unexpected way.

What I'd like is for the server to accept `application/json` _in addition_ to the custom media type, which is perfectly in line with HTTP content negotiation.

In other words, if my endpoint _accepts_ `application/vnd.something.crazy+json`, but I send plain `application/json`, I don't wan't to get `415 Unsupported Media Type`, since either way, _it's just JSON_. I'd except the server to say "hm, it looks like it's JSON, let me at least try to parse and validate it". 

However, if I send `text/plain` or `application/octet-stream`, then I would totally expect the server to reject that based on the media type alone. Hell, even in the case of `text/plain/` we might at least try to parse it with JSON (as long as that is reasonably safe).

The real issue is (or should be) whether the document conforms to the schema. 

Now, schemas are stored with the custom media types, so with the current architecture, they are inseparable. And that's fine, as long as allowing `application/json` coming in doesn't break validation. We can still set the custom types on content we produce. 

This is perfectly in line with the [Robustness Principle](http://en.wikipedia.org/wiki/Robustness_principle): 

"Be conservative in what you send, be liberal in what you accept."

@automatthew & @dyoder what do you say?

**Comment by dyoder** [comment]
Closing. The bug here is #18, since you can already specify `application/json` just fine.

**Comment by PandaWhisperer** [comment]
Not really. What I'm proposing here is to accept requests with a mime type of `application/json` _in addition_ to the custom mime type, as long as the body passes the schema validation. 

I feel like this has some merit on its own, so I'd like your opinion on whether that's a good idea, and if not, why not.
